libwallet/src/swap/ethereum/api.rs

```diff
@@ -0,0 +1,661 @@
+		swap: &Swap,
+	) -> Result<SwapTransactionsConfirmations, ErrorKind> {
+		let mwc_tip = self.node_client.get_chain_tip()?.0;
+
+		let is_seller = swap.is_seller();
+
+		let mwc_lock_conf =
+			self.get_slate_confirmation_number(&mwc_tip, &swap.lock_slate, !is_seller)?;
+		let mwc_redeem_conf =
+			self.get_slate_confirmation_number(&mwc_tip, &swap.redeem_slate, is_seller)?;
+		let mwc_refund_conf =
+			self.get_slate_confirmation_number(&mwc_tip, &swap.refund_slate, !is_seller)?;
+
+		// check eth transaction status
+		let secondary_tip = self.eth_height()?;
+		let mut secondary_lock_conf = None;
+		let eth_data = swap.secondary_data.unwrap_eth()?;
+		let secondary_lock_amount =
+			match self.eth_swap_details(eth_data.address_from_secret.clone()) {
+				Ok((refund_time, _, participant, value)) => {
+					let eth_data = swap.secondary_data.unwrap_eth()?;
+					//todo check locked eth money, 280 eth blocks ≈  1 hour
+					if (eth_data.redeem_address.clone().unwrap() == participant)
+						&& (refund_time > secondary_tip + 280u64)
+						&& value == swap.secondary_amount
+						&& swap.redeem_public.is_some()
+					{
+						let public_key = swap.redeem_public.clone().unwrap();
+						// convert mwc public key to ethereum public key format
+						let pub_key_array = public_key.0 .0;
+						let first_part: Vec<u8> = pub_key_array[..pub_key_array.len() / 2]
+							.to_owned()
+							.iter()
+							.rev()
+							.cloned()
+							.collect();
+						let second_part: Vec<u8> = pub_key_array[pub_key_array.len() / 2..]
+							.to_owned()
+							.iter()
+							.rev()
+							.cloned()
+							.collect();
+						let pub_key_vec: Vec<u8> = first_part
+							.into_iter()
+							.chain(second_part.into_iter())
+							.collect();
+						let wallet =
+							EthereumWallet::from_public_key(to_hex(&pub_key_vec).as_str()).unwrap();
+						let address = to_eth_address(wallet.address.clone().unwrap()).unwrap();
+						if eth_data.address_from_secret.unwrap() == address {
+							secondary_lock_conf = Some(1u64);
+						}
+					}
+					value
+				}
+				_ => 0,
+			};
+
+		let secondary_redeem_conf = match self.check_eth_transaction_confirmed(eth_data.redeem_tx) {
+			Ok(res) => Some(res),
+			_ => None,
+		};
+		let secondary_refund_conf = match self.check_eth_transaction_confirmed(eth_data.refund_tx) {
+			Ok(res) => Some(res),
+			_ => None,
+		};
+
+		Ok(SwapTransactionsConfirmations {
+			mwc_tip,
+			mwc_lock_conf,
+			mwc_redeem_conf,
+			mwc_refund_conf,
+			secondary_tip,
+			secondary_lock_conf,
+			secondary_lock_amount,
+			secondary_redeem_conf,
+			secondary_refund_conf,
+		})
+	}
+
+	/// Check How much ETH coins are locked on the chain
+	/// Return output with at least 1 confirmations because it is needed for refunds or redeems. Both party want to take everything
+	/// Return: (<pending_amount>, <confirmed_amount>, <least_confirmations>)
+	fn request_secondary_lock_balance(
```

Comment:
Do you handle coins in memory pool?  It is not important for p2p swap trades but is it important for marketplace because we need to detect if funds locked as fast as possible.

Reply:
Every operation(including initiate, redeem, refund) in eth is call in sync method which means when the funcion return the transaction was already confirmed on the chain. So  there is no coins in memory pool.

Reply:
As ethereum block period is about 15s, so i think we don't need consider coins in mem pool.